pkg/apis/feature/features.go

```diff
@@ -102,6 +102,26 @@ func (e Flags) String() string {
 	return fmt.Sprintf("%+v", map[string]Flag(e))
 }
 
+func (e Flags) NodeSelector() map[string]string {
+	// Check if NodeSelector is not nil
+	if e == nil {
+		return map[string]string{}
+	}
+
+	nodeSelectorMap := make(map[string]string)
+
+	for k, v := range e {
+		if strings.Contains(k, NodeSelectorLabel) {
+			key := strings.TrimPrefix(k, NodeSelectorLabel)
+			fmt.Println("value of k is",key)
```

Comment:
```suggestion
```

Also just FYI, we normally don't use `fmt.Println` for logging, you can get a logger by calling `logging.FromContext(ctx)`, and then pass the logger to where you need to log info.

Reply:
oh I dint wanted to print anything thanks for bringing this up